src/main/java/com/github/codeidoscope/ServerConnection.java

```diff
@@ -0,0 +1,58 @@
+package com.github.codeidoscope;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class ServerConnection implements ServerConnectionWrapper {
```

Comment:
It feels like that `Wrapper` in the name of an `Interface` is misleading as an interface isn't a wrapper.

You could have:
```
ServerConnectionWrapper s = new ServerConnection();
```

I think it's broadly used to name interface `ServerConnectionInterface`, which I think would be better but not ideal, maybe `Server` or `EchoServer` could do. What do you think?